Get-NetView.psm1

```diff
@@ -1131,6 +1131,195 @@ function MellanoxDetail {
+        uint data_mask_set = 0;
+
+        if ((dev_path == null) || (file_path == null))
+        {
+            Console.Write("Input pointers are NULL \n");
+            return 0;
+        }
+
+        local_path = local_path.TrimEnd('\\');
+        if (!local_path.StartsWith(@"\\?\Global\"))
+            local_path = @"\\?\Global\" + local_path;
+
+        if (1 == MarvellChipCode)
+        {
+            data_mask_set = B100_IHV_COLLECT_MASK;
+            ioctl_value   = B100_IOC_GET_IHV_DIAGNOSTIC_DATA;
+            local_path    = local_path + "#{5966d73c-bc2c-49b8-9315-c64c9919e976}";
+        }
+        else
+        {
+            data_mask_set = B10_IHV_COLLECT_MASK;
+            ioctl_value = B10_IOC_GET_IHV_DIAGNOSTIC_DATA;
+            local_path = local_path + "#{ea22615e-c443-434f-9e45-c4e32d83e97d}";
+        }
+
+        file_path = file_path + "\\";
+        file_path = file_path + filename;
+
+        SafeFileHandle shwnd = CreateFile(local_path, FileAccess.Write | FileAccess.Read, FileShare.Read | FileShare.Write, IntPtr.Zero, FileMode.Open, FileAttributes.Normal, IntPtr.Zero);
+        if (shwnd.IsClosed | shwnd.IsInvalid)
+        {
+            Exception e = Marshal.GetExceptionForHR(Marshal.GetLastWin32Error());
+            Console.Write("Error create file " + Marshal.GetLastWin32Error());
+            return 0;
+        }
+
+        byte[] OutBufptr = new byte[bytesSize];
+        Array.Clear(OutBufptr, 0, OutBufptr.Length);
+
+        DiagInput_t check = new DiagInput_t
+        {
+            revision = IHV_DIAG_REVISION,
+            data_mask = data_mask_set
+        };
+
+        result = DeviceIoControl(shwnd, ioctl_value, ref check, Marshal.SizeOf(check), OutBufptr, bytesSize, ref bytesReturned, IntPtr.Zero);
+        if (false == result)
+        {
+            Exception e = Marshal.GetExceptionForHR(Marshal.GetLastWin32Error());
+            Console.Write("Error DeviceIoControl " + Marshal.GetLastWin32Error());
+        }
+
+        FileStream file = File.Create(file_path);
+        file.Write(OutBufptr, 0, bytesReturned);
+        file.Close();
+
+        shwnd.Close();
+
+        if (result)
+            return bytesReturned;
+
+        return 0;
+    }
+}
+"@
+
+    $MarvellChipCode = 0;
+    $MarvellVbdService = $null;
+    $MarvellNicService = $null;
+
+    $pciId = (Get-NetAdapterAdvancedProperty -Name $NicName -AllProperties -RegistryKeyword "ComponentID").RegistryValue
+    $Marvell_pciId = $pciId.split("\")[0]
+    if ( $Marvell_pciId -eq "EBDRV"){
+        $MarvellChipCode = 0;
+        $MarvellVbdService = "ebdrv";
+        $MarvellNicService = "l2nd";
+    }
+    else{
+        $MarvellChipCode = 1;
+        $MarvellVbdService = "qebdrv";
+        $MarvellNicService = "l2nd2";
+    }
+
+    # Collect Marvell related event logs and miscellaneous details
+    $file = "$NicName-Bus.txt"
+    [String []] $cmds = "Get-EventLog -Source $MarvellVbdService -LogName System | Format-List"
```

Comment:
Use `Get-WinEvent` instead for PowerShell Core compat.

Reply:
Done.